lighthouse-core/gather/gatherers/console-messages.js

```diff
@@ -31,6 +31,22 @@ function remoteObjectToString(obj) {
 }
 
 class ConsoleMessages extends Gatherer {
+  /**
+   * @param {LH.Artifacts.ConsoleMessage} entry
+   * @return {LH.Audit.Details.SourceLocationValue | undefined}
+   */
+  static createSourceLocation(entry) {
```

Comment:
idk, it doesn't seem so bad to me. the logic for a genericly used "this is how you transform my results into useful audit details" doesn't really have a great place to live at the moment and living close to the gatherer implementation that produces that input is reasonable.

was there a specific alternative you had in mind?

Reply:
> idk, it doesn't seem so bad to me. the logic for a genericly used "this is how you transform my results into useful audit details" doesn't really have a great place to live at the moment and living close to the gatherer implementation that produces that input is reasonable.
> 
> was there a specific alternative you had in mind?

yeah, my instinct is to keep audit stuff out of gatherers because the more crossover there is the harder it is to reason about them separately, but it's a fair argument for this.

- Inline instead of a function call is one option (the transformation isn't a complicated one)
- Another is `audit.js`, where we've put other functions like this. It might be a good place for this too so that it's clear when these kinds of utilities are reaching critical mass and we really do need a good place (or places) for them to live

If folks disagree let's just drop it, this isn't a big deal, but it's also worth looking ahead to the mentioned #10930, at which point this audit details will need `ConsoleMessages` _and_ `SourceMaps` and `ScriptElements` to do the source map lookup and it becomes a bigger deal :)

Reply:
heh, the combo of sourcemaps *and* scriptelements being necessary is definitely a -1 for living on `ConsoleMessages`, audit.js SGTM :)